### PR TITLE
Add a first party share plugin

### PR DESCRIPTION
**Pull request by xster**
Trying my hand at making a plugin.
https://github.com/flutter/flutter/issues/9816

### CODE REVIEW

File: packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  @Override
+  public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+    if (call.method.equals("share")) {
+      if (!(call.arguments instanceof String)) {
+        result.error("ARGUMENT_ERROR", "String argument expected", null);
+        return;
+      }
+      final String text = (String) call.arguments;
+
```
Comment (szakarias) [inline packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java]: nit: you could consider moving lines 42-46 out in another method to separate our framework code from the platform-specific code.

File: packages/share/ios/Classes/SharePlugin.m [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- (instancetype)initWithController:
+    (FlutterViewController *)controller {
+  FlutterMethodChannel *shareChannel = [FlutterMethodChannel
+      methodChannelWithName:PLATFORM_CHANNEL
+            binaryMessenger:controller];
+
+  [shareChannel setMethodCallHandler:^(FlutterMethodCall *call,
+                                       FlutterResult result) {
+    if ([@"share" isEqualToString:call.method]) {
+      UIActivityViewController *activityViewController =
```
Comment (szakarias) [inline packages/share/ios/Classes/SharePlugin.m]: Similary here as for SharePlugin.java, you could consider moving out the platform-specific code.

**Comment by xster** [comment]
Added license to all example source files too

File: packages/share/android/build.gradle [hunk 3]
```diff
@@ -0,0 +1,32 @@
+group 'com.yourcompany.share'
```
Comment (xster) [inline packages/share/android/build.gradle]: done

File: packages/share/example/android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.yourcompany.share_example"
```
Comment (xster) [inline packages/share/example/android/app/src/main/AndroidManifest.xml]: done

File: packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java (hunk 1 above)
Comment (xster) [inline packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java]: makes sense, done

File: packages/share/ios/Classes/SharePlugin.m (hunk 2 above)
Comment (xster) [inline packages/share/ios/Classes/SharePlugin.m]: done

**Comment by szakarias** [comment]
It seems you missed to update the package names for the android part of the example app.

**Pull request merged**